package/src/components/Channel/Channel.tsx

```diff
@@ -882,6 +910,22 @@ const ChannelWithContext = <
       const oldListTopMessageId = messages[0]?.id;
       const oldListBottomMessage = messages[messages.length - 1];
 
+      const parseMessage = (message: typeof oldListTopMessage) =>
+        ({
+          ...message,
+          created_at: message.created_at.toString(),
+          pinned_at: message.pinned_at?.toString(),
+          updated_at: message.updated_at?.toString(),
+        } as unknown as MessageResponse<At, Ch, Co, Me, Re, Us>);
+
+      const failedMessages = messages
+        .filter((message) => message.status === 'failed')
+        .map(parseMessage);
+
+      const failedThreadMessages = thread
+        ? threadMessages.filter((message) => message.status === 'failed').map(parseMessage)
+        : [];
+
```

Comment:
This can go after line 944 I guess? Early returns should get priority over setting these values.